File: example/env.rs
fn main() {
    scope("str");
}

fn scope(a: &str) {
    let a = "local";
    println!("{}", a);
    
    let b = "outer";
    {
        let b = b;
        println!("{}", b);
    }
    
    let c = "global";
    {
        let show_c = || {
            println!("{}", c);
        };
        
        show_c();
        let c = "block";
        show_c();
    }
}

// Output:
// local
// outer
// global
// global

File: twi/example/debug/other_languages/env.py
a = "global"
def block():
    def show_a():
        print(a)
    
    show_a()
    a = "block"
    show_a()

block()

# [line 4] NameError: cannot access free variable 'a' where it is not associated with a value in enclosing scope
